cg/models/demultiplex/run_parameters.py

```diff
@@ -89,15 +97,77 @@ def __repr__(self):
         )
 
 
+class RunParametersHiSeq(RunParameters):
+    """Specific class for parsing run parameters of HiSeq2500 sequencing."""
+
+    def validate_instrument(self) -> None:
```

Comment:
Maybe this can be moved into the base class or a helper method, I think it is duplicated for each sequencer otherwise? Can look like
```python
class RunParameters:
    def validate_instrument(self, application_name: str) -> None:
        """Validate if the correct file was used to instantiate the class.""
        application = self.tree.find(RunParametersXMLNodes.APPLICATION_NAME)
        if application is None or application.text != application_name:
            raise RunParametersError(
                f"The file parsed does not correspond to the expected instrument: {application_name}"
            )

class RunParametersHiSeq(RunParameters):
    def validate_instrument(self) -> None:
        """Validate if a HiSeq file was used to instantiate the class."""
        super().validate_instrument(RunParametersXMLNodes.HISEQ_APPLICATION)
```

Reply:
I think this is out of scope as it would require a bit of work, especially as the node in the NovaSeqX is called differently

Reply:
I did not implement your suggestion because I could not make it work using the function inside the `__init__` function, but I did something that removes the duplicity of the function